services/bots/src/github-webhook/handlers/validate-cla.ts

```diff
@@ -0,0 +1,284 @@
+import { ConfigService } from '@nestjs/config';
+import { BaseWebhookHandler } from './base';
+
+import { EventPayloadMap } from '@octokit/webhooks-types';
+
+import { DynamoDB } from 'aws-sdk';
+
+const ignoredAuthors: Set<string> = new Set([
+  // Ignore bot accounts that are not masked as bots
+  'travis@travis-ci.org',
+  'ImgBotHelp@gmail.com',
+  'support@lokalise.com',
+]);
+
+const ignoredRepositories: Set<string> = new Set([
+  // Ignore repositories that do not contain code
+  'home-assistant/home-assistant.io',
+  'home-assistant/assets',
+]);
+
+const botContextName = 'cla-bot';
+
+enum IssueLabel {
+  CLA_ERROR = 'cla-error',
+  CLA_NEEDED = 'cla-needed',
+  CLA_RECHECK = 'cla-recheck',
+  CLA_SIGNED = 'cla-signed',
+}
+
+export class ValidateCla extends BaseWebhookHandler {
+  private ddbClient: DynamoDB;
+  private signersTableName: string;
+  private pendingSignersTableName: string;
+
+  constructor(configService: ConfigService) {
+    super(configService);
+    this.ddbClient = new DynamoDB({ region: configService.get('dynamodb.cla.region') });
+    this.signersTableName = configService.get('dynamodb.cla.signersTable');
+    this.pendingSignersTableName = configService.get('dynamodb.cla.pendingSignersTable');
+  }
+
+  async handle(eventType: string, payload: Record<string, any>) {
+    if (
+      ![
+        'pull_request.labeled',
+        'pull_request.opened',
+        'pull_request.reopened',
```

Comment:
We currently do not run on reopened events, but should we add that so the bot can be triggered by users that can not add the recheck label?

Reply:
yeah that's a good idea.